src/m365/spo/commands/tenant/tenant-applicationcustomizer-get.spec.ts

```diff
@@ -0,0 +1,364 @@
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+    auth.service.spoUrl = undefined;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.TENANT_APPLICATIONCUSTOMIZER_GET);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('fails validation if the id is not a valid GUID', async () => {
+    const actual = await command.validate({ options: { id: 'abc' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if the clientSideComponentId is not a valid GUID', async () => {
+    const actual = await command.validate({ options: { clientSideComponentId: 'abc' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation when all options are specified', async () => {
+    const actual = await command.validate({
+      options: {
+        title: title,
+        id: id,
+        clientSideComponentId: clientSideComponentId
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation when no options are specified', async () => {
+    const actual = await command.validate({
+      options: {
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation when title and id options are specified', async () => {
+    const actual = await command.validate({
+      options: {
+        title: title,
+        id: id
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation when title and clientSideComponentId options are specified', async () => {
+    const actual = await command.validate({
+      options: {
+        title: title,
+        clientSideComponentId: clientSideComponentId
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation when id and clientSideComponentId options are specified', async () => {
+    const actual = await command.validate({
+      options: {
+        id: id,
+        clientSideComponentId: clientSideComponentId
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if id is a valid GUID', async () => {
+    const actual = await command.validate({ options: { id: id } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passed validation when title specified', async () => {
+    const actual = await command.validate({ options: { title: title } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation if clientSideComponentId is valid', async () => {
+    const actual = await command.validate({ options: { clientSideComponentId: clientSideComponentId } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('throws error when tenant app catalog doesn\'t exist', async () => {
```

Comment:
We're missing a test case when retrieving a tenant app catalog fails with an exception. Let's add it here so that we can verify that we're handling it properly.